qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -310,6 +315,9 @@ def close_tab(self, tab, *, add_undo=True, new_undo=True):
             elif last_close == 'default-page':
                 self.openurl(config.val.url.default_page, newtab=True)
 
+        if not self.shutting_down:
```

Comment:
This make it so it isn't marked dirty when closing a window too. I should be checking if the whole application is shutting down, not just the current window. Would saving a reference to `app` in TabbedBrowser be bad?